VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/ImplementEqualsAndGetHashCode/ImplementEqualsAndGetHashCodeAnalyzer.cs

```diff
@@ -0,0 +1,152 @@
+        private static readonly string Message = VSDiagnosticsResources.ImplementEqualsAndGetHashCodeAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.ImplementEqualsAndGetHashCodeAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule =>
+            new DiagnosticDescriptor(DiagnosticId.ImplementEqualsAndGetHashCode, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) =>
+                context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration);
+
+        private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            var objectSymbol = context.SemanticModel.Compilation.GetSpecialType(SpecialType.System_Object);
+            IMethodSymbol objectEquals = null;
+            IMethodSymbol objectGetHashCode = null;
+
+            foreach (var symbol in objectSymbol.GetMembers())
+            {
+                if (!(symbol is IMethodSymbol))
+                {
+                    continue;
+                }
+
+                var method = (IMethodSymbol)symbol;
+                if (method.MetadataName == nameof(Equals) && method.Parameters.Count() == 1)
+                {
+                    objectEquals = method;
+                }
+
+                if (method.MetadataName == nameof(GetHashCode) && !method.Parameters.Any())
+                {
+                    objectGetHashCode = method;
+                }
+            }
+
+            if (context.Node is ClassDeclarationSyntax)
+            {
+                var classDeclaration = (ClassDeclarationSyntax) context.Node;
+
+                if (MembersDoNotContainOverridenEqualsAndGetHashCode(context.SemanticModel, classDeclaration.Members, objectEquals, objectGetHashCode) &&
+                    MembersContainNonStaticFieldOrProperty(classDeclaration.Members))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Rule, classDeclaration.Identifier.GetLocation(), classDeclaration.Identifier));
+                }
+            }
+            else
+            {
+                var structDeclaration = (StructDeclarationSyntax)context.Node;
+
+                if (MembersDoNotContainOverridenEqualsAndGetHashCode(context.SemanticModel, structDeclaration.Members, objectEquals, objectGetHashCode) &&
+                    MembersContainNonStaticFieldOrProperty(structDeclaration.Members))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Rule, structDeclaration.Identifier.GetLocation(), structDeclaration.Identifier));
+                }
+            }
+        }
+
+        private bool MembersDoNotContainOverridenEqualsAndGetHashCode(SemanticModel model, SyntaxList<SyntaxNode> members, IMethodSymbol objectEquals, IMethodSymbol objectGetHashCode)
```

Comment:
I need to calculate the method symbol on all overridden methods in the class, so passing in a single `IMethodSymbol` won't work.